File: problems/HackerRank/missingNumbers.ts
function missingNumbers(arr: number[], brr: number[]): number[] {
  arr.sort((a, b) => a - b);
  brr.sort((a, b) => a - b);

  let set = new Set();
  let counter = 0;

  for (let i = 0; i < brr.length; i++) {
    if (arr[counter] !== brr[i]) {
      set.add(brr[i]);
      continue;
    }
    counter++;
  }

  return Array.from(set.values()) as number[];
}

console.log(missingNumbers([7, 2, 5, 3, 5, 3], [7, 2, 5, 4, 6, 3, 5, 3]));

File: problems/HackerRank/week5/missingNumbers.py
def missingNumbers(arr, brr):
    dic = {}
    ans = []
    
    for ele in brr:
        dic[ele] = dic.get(ele, 0) + 1
    
    for ele in arr:
        dic[ele] = dic.get(ele) - 1
        
    for ele in dic:
        if dic.get(ele) > 0:
            ans.append(ele)
    
    ans.sort()
    
    return ans 


def missingNumbers2(arr, brr):
    arr.sort()
    brr.sort()
    s = set()
    
    counter = 0
    for i in range(len(brr)):
        if counter >= len(arr) or arr[counter] != brr[i]:
            s.add(brr[i])
            continue
        counter += 1

    return sorted(s)